tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+            chk.add_arg(f"i{iteration}_actual_byte_rate", actual_byte_rate)
+            chk.add_arg("produce_byte_rate", self.params.produce_byte_rate)
+            chk.add_arg("producer_tolerance", self.params.producer_tolerance)
+            chk.add_check(
+                "Check the workload is respecting rate limit",
+                f"i{iteration}_actual_byte_rate < "
+                "produce_byte_rate * producer_tolerance")
+
+            # Conduct message checks
+            self.logger.info(
+                f"Iteration {iteration}; calculating stats for all partitions")
+            stats = self._calculate_statistic(self.topics, self.rpk,
+                                              self.redpanda)
+            # Ensure that all messages made it to topic
+            self.logger.info(f"calculated hwm: {stats['hwm']}, "
+                             f"message count: {self.params.msg_count}")
+            chk.add_arg(f"i{iteration}_sum_high_watermarks", stats['hwm'])
+            chk.add_arg(f"i{iteration}_msg_count",
+                        self.params.msg_count * iteration)
+            chk.add_check(
+                "Ensure that all messages made it to topic",
+                f"i{iteration}_sum_high_watermarks >= i{iteration}_msg_count",
+            )
+            self.logger.info(
+                f"Upload counts: {stats['manifest_uploads']} manifests, "
+                f"{stats['segment_uploads']} segments")
+            chk.add_arg(f"i{iteration}_manifest_uploads",
+                        stats['manifest_uploads'])
+            chk.add_arg(f"i{iteration}_segment_uploads",
+                        stats['segment_uploads'])
+            chk.add_check(
+                f"Iteration {iteration}; non-zero manifest uploads",
+                f"i{iteration}_manifest_uploads > 0",
+            )
+            chk.add_check(
+                f"Iteration {iteration}; non-zero segment uploads",
+                f"i{iteration}_segment_uploads > 0",
+            )
+            checks_duration = time.time() - checks_start
+            scrub_start = time.time()
+            # Run 'rp_storage_tool' to check for anomalies
+            try:
+                self.redpanda.stop_and_scrub_object_storage(run_timeout=1800)
+            except RuntimeError as exc:
+                scrub_exception = exc
+            scrub_duration = time.time() - scrub_start
+
+            iteration_timings["produce"][iteration] = produce_duration
+            iteration_timings["high_watermark"][iteration] = stats['hwm']
+            iteration_timings["manifest_uploads"][iteration] = stats[
+                'manifest_uploads']
+            iteration_timings["segment_uploads"][iteration] = stats[
+                'segment_uploads']
+            iteration_timings["checks"][iteration] = checks_duration
+            iteration_timings["scrub"][iteration] = scrub_duration
+            iteration_timings["sp_mn_uploads_total"][iteration] = stats[
```

Comment:
It's not obvious where this value is being inspected later?  For an infinite retention test, verifying that we are generating some number of spillover manifests should  be an important success condition.